chaco/barplot.py

```diff
@@ -415,8 +415,8 @@ def _update_mappers(self):
         self.invalidate_draw()
         self._cache_valid = False
 
-    @on_trait_change('line_color, line_width, fill_color, alpha')
-    def _attributes_changed(self):
+    @observe('line_color, line_width, fill_color, alpha')
+    def _attributes_changed(self, event):
```

Comment:
i'm not a fan of the `*_changed` method name especially when it's not a trait change handler but i can't think of one better either.

Reply:
In general, i've been told that the methods shouldn't be named based on what changed - rather, they should be named to convey what they are going to do.

Until we find a better name, maybe we can just call this `_attributes_updated` - which i've seen elsewhere in this PR.

Reply:
alternatively, we could rename it to `_invalidate_and_redraw` - if no such method already exists in the class heirarchy - because that's what the method does.